compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2821,7 +2838,11 @@ define_print_and_forward_display! {
     }
 
     ty::AliasTy<'tcx> {
-        p!(print_def_path(self.def_id, self.substs));
+        if let DefKind::Impl { of_trait: false } = cx.tcx().def_kind(cx.tcx().parent(self.def_id)) {
```

Comment:
I'm no longer using `AliasTy::kind()` here since it ICEs on `AssocConst`s. `AliasTy`s can actually be `AssocConst`s for associated const projections even though it's not properly documented anywhere as I assume it to be a temporary hack to make `#![feature(associated_const_equality)]` work until `AliasTerm` is introduced.